presto-native-execution/src/test/java/com/facebook/presto/nativeworker/TestHiveQueries.java

```diff
@@ -364,6 +364,110 @@ private Session enableDecimalParsing()
                 .build();
     }
 
+    @Test
+    public void testDecimalArithmetic() {
+        // Addition of two Long Decimal columns of type DECIMAL(35,20) and DECIMAL(29,4)
+        // and also contains NULLs.
+        assertQuery(
+                "SELECT n + m from (values " +
+                        "(CAST('999999999999999.999' as decimal(18,3)), CAST('1' as decimal(1,0)))," +
+                        "(CAST('-123456789012345.123456' as DECIMAL(21,6)), CAST('-9999999999999999' as DECIMAL(25, 0)))," +
+                        "(CAST('1.23' as DECIMAL(3,2)), CAST('-0.0005' as DECIMAL(5,4)))," +
+                        "(CAST('1.33333333333333333333' as DECIMAL(23,20)), CAST('-0.0005' as DECIMAL(5,4)))," +
+                        "(NULL, NULL), (decimal'1.23', NULL)) t(n, m)");
+
+        // Addition of two short decimal columns of type DECIMAL(10,7) and DECIMAL(10,5)
+        assertQuery("SELECT n + m from (values (decimal'1.1', decimal'-1.1')," +
+                "(decimal'-0.0000004', decimal'-0.12345')," +
+                "(decimal'123', decimal'13245')) t(n, m)");
+
+        // numeric limits
+        assertQueryFails("SELECT n + m from (values (CAST('99999999999999999999999999999999999999' as DECIMAL(38,0))," +
```

Comment:
Do we need cast? can we simply say `decimal '99999999999999999999999999999999999999'`?